tests/test_sycl_queue.py

```diff
@@ -2504,3 +2504,21 @@ def test_ediff1d(device, to_end, to_begin):
     res = dpnp.ediff1d(x, to_end=to_end, to_begin=to_begin)
 
     assert_sycl_queue_equal(res.sycl_queue, x.sycl_queue)
+
+
+@pytest.mark.parametrize(
+    "device_0",
+    valid_devices,
+    ids=[device.filter_string for device in valid_devices],
+)
+@pytest.mark.parametrize(
+    "device_1",
+    valid_devices,
+    ids=[device.filter_string for device in valid_devices],
+)
+def test_indices_sparse(device_0, device_1):
+    x0 = dpnp.array([0, 1], device=device_0)
+    x1 = dpnp.array([2, 4], device=device_1)
+    ixgrid = dpnp.ix_(x0, x1)
+    assert_sycl_queue_equal(ixgrid[0].sycl_queue, x0.sycl_queue)
```

Comment:
Although this test works, but having output on different devices does not allow to re-construct an array.

It is kind of similar to the discussion about data type of input sequences. Although the function works for all data types, but for its intended use case of this function, it only make sense to have integer/boolean dtypes. 
I do not think we need to change anything but it is good to be aware of it.

```
import dpnp
x0 = dpnp.array([0, 1], device="cpu")
x1 = dpnp.array([2, 4], device="gpu")
ixgrid = dpnp.ix_(x0, x1)

a = dpnp.arange(10).reshape(2, 5)
a[ixgrid]
# ExecutionPlacementError: ...
```